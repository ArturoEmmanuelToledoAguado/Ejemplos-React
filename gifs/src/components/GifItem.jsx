import PropTypes from 'prop-types';

export const GifItem = ({title,url}) => {
    const copyPath = ()=>{
        navigator.clipboard.writeText(url)
    }

  return (
    <div className="">
        <img 
            src={url}
            alt={title}
        />
        <p>{title}</p>
        <button className=""onClick={copyPath}>Copy Path</button>
    </div>
  )
}

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}