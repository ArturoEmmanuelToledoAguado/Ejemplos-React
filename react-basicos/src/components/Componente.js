import React from "react";
//import React,{Component} from "react";

const Componente=props=> <h2>{props.msg}</h2>

/* Componente en funcion declarada
function Componente(props) {
    return <h2>{props.msg}</h2>;
}*/

/* Componente con clase
class Componente extends Component{
    render() {
        return <h2>{this.props.msg}</h2>;
    }
}*/

export default Componente;