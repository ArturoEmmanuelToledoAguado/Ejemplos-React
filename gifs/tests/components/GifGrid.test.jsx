import { render, screen } from "@testing-library/react"
import { GifGrid } from "../../src/components/GifGrid"
import { useFetchGifs } from '../../src/hooks/useFetchGifs.js';

jest.mock('../../src/hooks/useFetchGifs.js')

describe('prueba sobre GifGrid', () => { 
    const category='batman'

    test('Esta prueba es para hacer match con el snapshot', () => { 
        useFetchGifs.mockReturnValue({
            images:[],
            isLoading:true
        })
        const {container} = render(<GifGrid category={category}/>)
        expect(container).toMatchSnapshot();
        //screen.debug()
     })

    test('Prueba del arreglo de imagenes', () => { 
        const image =[{
            id:'2',
            title:'Batman',
            url:'http:Github.com/'
        }]
        useFetchGifs.mockReturnValue({
            images:image,
            isLoading:false
        })
        render(<GifGrid category={category}/>)
        expect(screen.getAllByRole('img').length).toBe(1)
     })

    test('Valida la existencia de h2', () => { 
        render(<GifGrid category={category}/>)
        expect(screen.getAllByText(category)).toBeTruthy()//Valida que existe alguna etiqueta h2
     })
 })