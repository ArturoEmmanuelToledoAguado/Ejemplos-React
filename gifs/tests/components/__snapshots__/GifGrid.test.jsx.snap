// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba sobre GifGrid Esta prueba es para hacer match con el snapshot 1`] = `
<div>
  <h2>
    batman
  </h2>
  <h2>
    Loading...
  </h2>
</div>
`;
